xdsl/dialects/memref.py

```diff
@@ -509,6 +511,40 @@ class ExpandShapeOp(AlterShapeOp):
     name = "memref.expand_shape"
 
 
+@irdl_op_definition
+class ExtractStridedMetaDataOp(IRDLOperation):
+    name = "memref.extract_strided_metadata"
+
+    source: Operand = operand_def(MemRefType)
+
+    base_buffer: OpResult = result_def(MemRefType)
+    offset: OpResult = result_def(IndexType)
+    sizes: VarOpResult = var_result_def(IndexType)
+    strides: VarOpResult = var_result_def(IndexType)
+
+    irdl_options = [AttrSizedResultSegments()]
+
+    @staticmethod
+    def from_source_memref(
+        source: SSAValue | Operation, source_type: MemRefType[Attribute]
+    ):
```

Comment:
Can you make this an `__init__`? That's what we are trying to move towards in xDSL

Reply:
Is this an __init__ though? creating a bare op of this kind should not really include a source type, should it?

Reply:
I remember being torn by the very same thought on subview itself;
I propose you just do what's still done there, and @AntonLydike could file an issue or Zulip thread to discuss it independently?

Reply:
Hmmm, I don't know if I can follow. How else would you create this op if not from a source memref?

Reply:
Maybe you could clear this up by adding a docstring to the method saying exactly what it's for and what each operand does?

Reply:
May I ask why `source` and `source_type` are passed in separately? I'd expect `source` to come with a type